pulsar-client/src/test/java/org/apache/pulsar/client/impl/conf/ClientConfigurationDataTest.java

```diff
@@ -18,38 +18,102 @@
-        w = m.writer();
-    }
-
+    private static final String BASIC_AUTH_PLUGIN_CLASS = AuthenticationBasic.class.getName();
 
     @Test
     public void testDoNotPrintSensitiveInfo() throws JsonProcessingException {
         ClientConfigurationData clientConfigurationData = new ClientConfigurationData();
         clientConfigurationData.setTlsTrustStorePassword("xxxx");
         clientConfigurationData.setSocks5ProxyPassword("yyyy");
         clientConfigurationData.setAuthentication(new AuthenticationToken("zzzz"));
-        String s = w.writeValueAsString(clientConfigurationData);
-        Assert.assertFalse(s.contains("xxxx"));
-        Assert.assertFalse(s.contains("yyyy"));
-        Assert.assertFalse(s.contains("zzzz"));
+        ObjectMapper objectMapper = new ObjectMapper();
+        objectMapper.configure(SerializationFeature.FAIL_ON_EMPTY_BEANS, false);
+        String serializedConf = objectMapper.writeValueAsString(clientConfigurationData);
+        assertThat(serializedConf).doesNotContain("xxxx", "yyyy", "zzzz");
+    }
+
+    @Test
+    public void testGetAuthWhenSetFromPropsUsingAuthParams() {
+        ClientConfigurationData conf = new ClientConfigurationData();
+        conf.setAuthPluginClassName(BASIC_AUTH_PLUGIN_CLASS);
+        conf.setAuthParams(basicAuthParams("user1", "pass1"));
+        assertThatConfHasBasicAuth(conf, "user1", "pass1");
+    }
+
+    @Test
+    public void testGetAuthWhenSetFromPropsUsingAuthParamMap() {
+        ClientConfigurationData conf = new ClientConfigurationData();
+        conf.setAuthPluginClassName(BASIC_AUTH_PLUGIN_CLASS);
+        conf.setAuthParamMap(basicAuthParamMap("user1", "pass1"));
+        assertThatConfHasBasicAuth(conf, "user1", "pass1");
+    }
+
+    @Test
+    public void testGetAuthWhenSetFromPropsWithAllPropsAvailable() {
+        ClientConfigurationData conf = new ClientConfigurationData();
+        conf.setAuthPluginClassName(BASIC_AUTH_PLUGIN_CLASS);
+        conf.setAuthParams(basicAuthParams("user1", "pass1"));
+        conf.setAuthParamMap(basicAuthParamMap("user2", "pass2"));
+        assertThatConfHasBasicAuth(conf, "user1", "pass1");
+    }
+
+    private void assertThatConfHasBasicAuth(ClientConfigurationData conf, String userId, String password) {
+        assertThat(conf.getAuthentication())
+                .isInstanceOf(AuthenticationBasic.class)
+                .hasFieldOrPropertyWithValue("userId", userId)
+                .hasFieldOrPropertyWithValue("password", password);
+    }
+
+    @Test
+    public void testGetAuthWhenSetFromPropsWithNoPropsAvailable() {
+        ClientConfigurationData conf = new ClientConfigurationData();
+        assertThatConfHasNoAuth(conf);
+    }
+
+    @Test
+    public void testGetAuthWhenSetFromPropsWithOnlyPluginClassNameAvailable() {
+        ClientConfigurationData conf = new ClientConfigurationData();
+        conf.setAuthPluginClassName(BASIC_AUTH_PLUGIN_CLASS);
+        assertThatConfHasNoAuth(conf);
```

Comment:
It should be basic auth.

Reply:
In this case there are no `authParams` nor `authParamsMap` so it defaults to disabled. Unless I am missing something?

Reply:
Perhaps the authentication plugin doesn't care about authentication parameters.

Reply:
I see what you are saying now, but I fear that would cause quite the ripple in scope: 

* this is the same logic that was here before this code proposal
* AFAIK all of the current auth impls require at least one parameter. 
* At least some impls are not able to handle this case (eg AB will NPE)

Reply:
Ok